stable_baselines3/ppo/ppo.py

```diff
@@ -231,18 +233,27 @@ def train(self) -> None:
 
                 loss = policy_loss + self.ent_coef * entropy_loss + self.vf_coef * value_loss
 
+                # Calculate approximate form of reverse KL Divergence for early stopping
+                # see issue #417: https://github.com/DLR-RM/stable-baselines3/issues/417
+                # and Schulman blog: http://joschu.net/blog/kl-approx.html 
+                with th.no_grad():
+                    r = rollout_data.old_log_prob - log_prob
+                    approx_kl_div = th.mean((th.exp(r) - 1) - r).cpu().numpy()
+                    approx_kl_divs.append(approx_kl_div)
+
+                if self.target_kl is not None and approx_kl_div > 1.5 * self.target_kl:
+                    continue_training = False
+                    print(f"Early stopping at step {epoch} due to reaching max kl: {approx_kl_div:.2f}")
```

Comment:
Off-PR, but this should be inside `if self.verbose >= 1:` (a debug-print-ish thing)